python/tank/bootstrap/configuration.py

```diff
@@ -93,13 +93,18 @@ def get_tk_instance(self, sg_user):
         # and pass that into the factory method.
 
         # Previous versions of the PipelineConfiguration API didn't support having a descriptor
-        # passed in, so we'll have to be backwards compatible with these. If the pipeline
-        # configuration does support the get_configuration_descriptor method however, we can
-        # pass the descriptor in.
-        if hasattr(pipelineconfig.PipelineConfiguration, "get_configuration_descriptor"):
+        # passed in, so we'll have to be backwards compatible with these. We'll inspect the arguments
+        # that can be passed down to it to figure out which version of the API we're using.
+
+        args = inspect.getargspec(pipelineconfig.PipelineConfiguration.__init__)[0]
```

Comment:
Writing it out to `pipeline_configuration.yml` however is not in line with how we want to do things in the future however.

I'm not sure how we'd fix this tough. Are we bound to support that extra parameter forever now ? Yikes!